typings/styled-components.d.ts

```diff
@@ -1,7 +1,7 @@
 import * as React from "react";
-import { StatelessComponent, ComponentClass } from "react";
+import { StatelessComponent, ComponentClass, PureComponent } from "react";
 
-type Component<P> = ComponentClass<P> | StatelessComponent<P>;
+type Component<P> = ComponentClass<P> | StatelessComponent<P> | PureComponent<P, any>;
```

Comment:
I saw how withRouter was implemented in React Router, they were also using PureComponent. I'll remove this :)